ml4h/explorations.py

```diff
@@ -691,6 +699,191 @@ def infer_with_pixels(args):
+        header += ['mri_date']
+        inference_writer_true.writerow(header)
+        inference_writer_pred.writerow(header)
+
+        while True:
+            batch = next(generate_test)
+            data, labels, tensor_paths = batch[BATCH_INPUT_INDEX], batch[BATCH_OUTPUT_INDEX], batch[BATCH_PATHS_INDEX]
+            if tensor_paths[0] in tensor_paths_inferred:
+                next(generate_test)  # this print end of epoch info
+                logging.info(
+                    f"Inference on {stats_counter['count']} tensors finished. Inference TSV files at: {inference_tsv_true}, {inference_tsv_pred}",
+                )
+                break
+
+            img = data[tm_in.input_name()]
+            rescaled_img = tm_in.rescale(img)
+            y_pred = model.predict(data, batch_size=args.batch_size, verbose=0)
+            y_pred = np.argmax(y_pred, axis=-1)
+            y_pred = _to_categorical(y_pred, nb_orig_channels)
+            if args.analyze_ground_truth:
+                y_true = labels[tm_out.output_name()]
+
+            sample_id = os.path.basename(tensor_paths[0]).replace(TENSOR_EXT, '')
+            date = dates_dict[sample_id]
+
+            if args.analyze_ground_truth:
+                if do_intensity_thresh:
+                    y_true = _thresh_labels_above(y_true, img, args.intensity_thresh, intensity_thresh_in_channels, intensity_thresh_out_channel, nb_orig_channels)
+                y_true = np.delete(y_true, bad_channels, axis=-1)
+                if args.erosion_radius > 0:
+                    y_true = binary_erosion(y_true, structure).astype(y_true.dtype)
+                means_true, medians_true, stds_true = _compute_masked_stats(rescaled_img, y_true, nb_good_channels)
+                csv_row_true = _get_csv_row(sample_id, means_true, medians_true, stds_true, date)
+                inference_writer_true.writerow(csv_row_true)
+
+            if do_intensity_thresh:
+                y_pred = _thresh_labels_above(y_pred, img, args.intensity_thresh, intensity_thresh_in_channels, intensity_thresh_out_channel, nb_orig_channels)
+            y_pred = np.delete(y_pred, bad_channels, axis=-1)
+            if args.erosion_radius > 0:
+                y_pred = binary_erosion(y_pred, structure).astype(y_pred.dtype)
+            means_pred, medians_pred, stds_pred = _compute_masked_stats(rescaled_img, y_pred, nb_good_channels)
+            csv_row_pred = _get_csv_row(sample_id, means_pred, medians_pred, stds_pred, date)
+            inference_writer_pred.writerow(csv_row_pred)
+
+            tensor_paths_inferred.add(tensor_paths[0])
+            stats_counter['count'] += 1
+            if stats_counter['count'] % 250 == 0:
+                logging.info(f"Wrote:{stats_counter['count']} rows of inference.  Last tensor:{tensor_paths[0]}")
+
+    # Scatter plots
+    if args.analyze_ground_truth:
+        df_true = pd.read_csv(inference_tsv_true, delimiter='\t', quotechar='"', quoting=csv.QUOTE_MINIMAL)
```

Comment:
refactor this into a function just to keep things modular, arguably it belongs in `plots.py` but only if you think we might use it in another context.

Reply:
I refactored this into its own function, and agree it looks much cleaner now. But, it's very specific to this context so for now I think we can keep it in explorations.py